engine/classes/Elgg/ViewsService.php

```diff
@@ -485,6 +489,88 @@ public function unextendView($view, $view_extension) {
 	}
 
 	/**
+	 * Registers an AMD module as a view dependency
+	 * 
+	 * @param string $view   View name
+	 * @param string $module AMD module name
+	 * @return void
+	 * @access private
+	 */
+	public function addDep($view, $module) {
+		$view = $this->canonicalizeViewName($view);
+
+		if ($this->isCacheableView($view)) {
+			$this->logger->warn("Cacheable view '$view' can not depend on an AMD module");
```

Comment:
I also can't think of a reasonable use case for this. Is there a good reason not to throw exception? Warnings get ignored.